ext/socket/lib/socket.rb

```diff
@@ -593,6 +594,74 @@ def accept_nonblock(exception: true)
     __accept_nonblock(exception)
   end
 
+  def self.start_getaddrinfo_v6(host, port, queue, resolv_timeout)
+    return Thread.new do
+      ai_list = Addrinfo.getaddrinfo(host, port, :PF_INET6, :STREAM, timeout: resolv_timeout)
+      ai_list.each {|ai| queue.push(ai) }
+    rescue SocketError => ex
+      case ex.message
+      when "getaddrinfo: Address family for hostname not supported" # when IPv6 is not supported
+        # ignore
+      when "getaddrinfo: Temporary failure in name resolution" # when timed out (EAI_AGAIN)
+        # ignore
+      else
+        raise ex
+      end
+    rescue ClosedQueueError
+      # timed out. ignore exception
+    ensure
+      queue.push(:getaddrinfo_v6_done) rescue nil
```

Comment:
`queue.push` raises an exception if `queue` is already closed.
The exception terminates the thread and it is reported to the terminal.
(report_on_exception is true by default now)

Reply:
I fixed it to handle ClosedQueueError.